app/jobs/list_project_contents_job.rb

```diff
@@ -1,30 +1,38 @@
 # frozen_string_literal: true
 class ListProjectContentsJob < ApplicationJob
-  # Asynchronously query Mediaflux for Project assets and cache the results for rendering to the end-user
-  def perform(user_id:, project_title:)
-    # @todo Query for all contents within any given project
+  def perform(user_id:, project_id:)
+    project = Project.find(project_id)
+    raise "Invalid project id #{project_id} for job #{job_id}" if project.nil?
+    user = User.find(user_id)
+    raise "Invalid user id #{user_id} for job #{job_id}" if user.nil?
+    filename = filename_for_export
 
-    @user_id = user_id
-    raise(ArgumentError, "Failed to retrieve the User for the ID #{@user_id}") if user.nil?
+    # Queries Mediaflux for the file list and saves it to a CSV file.
+    project.file_list_to_file(session_id: mediaflux_session, filename: filename)
 
-    @project_title = project_title
-
-    user.user_jobs << user_job
-    user.save!
+    mark_user_job_as_complete(project: project, user: user)
   end
 
   private
 
-    def user
-      @user ||= User.find_by(id: @user_id)
+    def mediaflux_session
+      logon_request = Mediaflux::Http::LogonRequest.new
+      logon_request.resolve
+      logon_request.session_token
+    end
+
+    # TODO: This only works with one server. In a multi-server environment
+    # there is no guarantee that the client will be able to fetch the file
+    # since requests could go to any of the many servers.
+    # See https://github.com/pulibrary/tiger-data-app/issues/523
```

Comment:
Thank you for this documentation, this could definitely lead to in consistencies in results before the issue is resolved.